File: HTTPShortcuts/curl_command/src/test/java/ch/rmy/curlcommand/CommandParserTest.java
package ch.rmy.curlcommand;

import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertArrayEquals;

public class CommandParserTest {

    @Test
    public void testParser() {
        String target = "curl -X \"Hello 'World'\" \"escaped \\\" quotes and backslashes \\\\\" --bla 'foo'";
        String[] expected = new String[]{
                "curl", "-X", "Hello 'World'", "escaped \" quotes and backslashes \\", "--bla", "foo"
        };
        List<String> arguments = CommandParser.parseCommand(target);
        String[] actual = arguments.toArray(new String[arguments.size()]);
        assertArrayEquals(expected, actual);
    }

}

File: HTTPShortcuts/curl_command/src/test/kotlin/ch/rmy/curlcommand/CommandParserTest.kt
package ch.rmy.curlcommand

import kotlin.test.Test
import kotlin.test.assertEquals

class CommandParserTest {

    @Test
    fun testParser() {
        val target = "curl -X \"Hello 'World'\" \"escaped \\\" quotes and backslashes \\\\\" --bla 'foo'"
        val expected = listOf("curl", "-X", "Hello 'World'", "escaped \" quotes and backslashes \\", "--bla", "foo")
        val actual = CommandParser.parseCommand(target)
        assertEquals(expected, actual)
    }
}
